.demo/env/clouds/aws/demo/jobs.demo.yaml

```diff
@@ -0,0 +1,12 @@
+teamConfig:
+    teams:
+        demo:
+            jobs:
+                - name: job-hello-world
+                  enabled: true
```

Comment:
this should be default and then we drop the flag

Reply:
and it seems to make no sense to just run it on deployment imo, cause that is what this will do.

I think it makes sense to look at the options helm offers and maybe expose that here as configuration. Like always deleting and starting a job fresh, or only when the spec changed. I am not sure these options exist but please investigate what is possible for our users.

Reply:
> this should be default and then we drop the flag

If I understand correctly, if we add this to the scope then we should add that to the chart. 

> and it seems to make no sense to just run it on deployment imo, cause that is what this will do.

I think it makes sense to look at the options helm offers and maybe expose that here as configuration. Like always deleting and starting a job fresh, or only when the spec changed. I am not sure these options exist but please investigate what is possible for our users.

I'll add that to the scope as well. It makes sense to me to explore that yes.

Reply:
If so, I've updated the [spec](https://github.com/redkubes/otomi-core/issues/204), please verify if this is the intended scope.

Reply:
> and it seems to make no sense to just run it on deployment imo, cause that is what this will do.

_Note: I am exploring options to run it on spec change._

I just ran the command from the perspective of an end-user; it will run on deployment, yes. But having made no changes to the spec, it will still run a new one (see output): 

```
otomi apply -l name=jobs-demo
Comparing release=jobs-demo, chart=../charts/jobs
	Release was not present in Helm.  Diff will show entire contents as new.
team-demo, 0-888cdb3c-5da2-4c50-9779-9b45f689a053, Job (batch) has been added:
-
+ # Source: jobs/templates/job.yaml
+ apiVersion: batch/v1
+ kind: Job
+ metadata:
+   name: 0-888cdb3c-5da2-4c50-9779-9b45f689a053
+   labels:
+     helm.sh/chart: jobs-0.1.0
+     app.kubernetes.io/name: jobs
+     app.kubernetes.io/instance: jobs-demo
+     app.kubernetes.io/version: "0.1.0"
+     app.kubernetes.io/managed-by: Helm
+ spec:
+   template:
+     metadata:
+       labels:
+         helm.sh/chart: jobs-0.1.0
+         app.kubernetes.io/name: jobs
+         app.kubernetes.io/instance: jobs-demo
+         app.kubernetes.io/version: "0.1.0"
+         app.kubernetes.io/managed-by: Helm
+       annotations:
+         checksum/secret: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546
+         checksum/config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546
+     spec:
+       serviceAccountName: default
+       containers:
+       - image: "busybox:latest"
+         imagePullPolicy: "IfNotPresent"
+         name: "0"
+         command: ["sh", "-c"]
+         args:
+           - |
+             echo "Hello World"
+         resources:
+           limits:
+             cpu: 200m
+             memory: 256Mi
+           requests:
+             cpu: 100m
+             memory: 128Mi
+         envFrom:
+         volumeMounts:
+           - name: fakeroot
+             mountPath: /fakeroot.pem
+             subPath: fakeroot.pem
+       restartPolicy: Never
+       volumes:
+         - name: fakeroot
+           configMap:
+             name: fakeroot
+   backoffLimit: 3
+   ttlSecondsAfterFinished: 86400
Upgrading release=jobs-demo, chart=../charts/jobs
Release "jobs-demo" does not exist. Installing it now.
NAME: jobs-demo
LAST DEPLOYED: Tue Dec 22 11:25:04 2020
NAMESPACE: team-demo
STATUS: deployed
REVISION: 1
TEST SUITE: None
Listing releases matching ^jobs-demo$
jobs-demo	team-demo	1       	2020-12-22 11:25:04.762810766 +0000 UTC	deployed	jobs-0.1.0	0.1.0
UPDATED RELEASES:
NAME        CHART            VERSION
jobs-demo   ../charts/jobs     0.1.0
 sebastiaanverbeek@mbprosv  ~/Projects/redkubes/dev/otomi-core   task/204/spec-enabled ± otomi apply -l name=jobs-demo
Comparing release=jobs-demo, chart=../charts/jobs
team-demo, 0-e4c9d60c-8522-4a1f-b5d6-efc9a0a20cd4, Job (batch) has been removed:
- # Source: jobs/templates/job.yaml
- apiVersion: batch/v1
- kind: Job
- metadata:
-   name: 0-e4c9d60c-8522-4a1f-b5d6-efc9a0a20cd4
-   labels:
-     helm.sh/chart: jobs-0.1.0
-     app.kubernetes.io/name: jobs
-     app.kubernetes.io/instance: jobs-demo
-     app.kubernetes.io/version: "0.1.0"
-     app.kubernetes.io/managed-by: Helm
- spec:
-   template:
-     metadata:
-       labels:
-         helm.sh/chart: jobs-0.1.0
-         app.kubernetes.io/name: jobs
-         app.kubernetes.io/instance: jobs-demo
-         app.kubernetes.io/version: "0.1.0"
-         app.kubernetes.io/managed-by: Helm
-       annotations:
-         checksum/secret: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546
-         checksum/config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546
-     spec:
-       serviceAccountName: default
-       containers:
-       - image: "busybox:latest"
-         imagePullPolicy: "IfNotPresent"
-         name: "0"
-         command: ["sh", "-c"]
-         args:
-           - |
-             echo "Hello World"
-         resources:
-           limits:
-             cpu: 200m
-             memory: 256Mi
-           requests:
-             cpu: 100m
-             memory: 128Mi
-         envFrom:
-         volumeMounts:
-           - name: fakeroot
-             mountPath: /fakeroot.pem
-             subPath: fakeroot.pem
-       restartPolicy: Never
-       volumes:
-         - name: fakeroot
-           configMap:
-             name: fakeroot
-   backoffLimit: 3
-   ttlSecondsAfterFinished: 86400
+
team-demo, 0-fe85431a-1f1a-422d-bf44-f6fe58f53628, Job (batch) has been added:
-
+ # Source: jobs/templates/job.yaml
+ apiVersion: batch/v1
+ kind: Job
+ metadata:
+   name: 0-fe85431a-1f1a-422d-bf44-f6fe58f53628
+   labels:
+     helm.sh/chart: jobs-0.1.0
+     app.kubernetes.io/name: jobs
+     app.kubernetes.io/instance: jobs-demo
+     app.kubernetes.io/version: "0.1.0"
+     app.kubernetes.io/managed-by: Helm
+ spec:
+   template:
+     metadata:
+       labels:
+         helm.sh/chart: jobs-0.1.0
+         app.kubernetes.io/name: jobs
+         app.kubernetes.io/instance: jobs-demo
+         app.kubernetes.io/version: "0.1.0"
+         app.kubernetes.io/managed-by: Helm
+       annotations:
+         checksum/secret: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546
+         checksum/config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546
+     spec:
+       serviceAccountName: default
+       containers:
+       - image: "busybox:latest"
+         imagePullPolicy: "IfNotPresent"
+         name: "0"
+         command: ["sh", "-c"]
+         args:
+           - |
+             echo "Hello World"
+         resources:
+           limits:
+             cpu: 200m
+             memory: 256Mi
+           requests:
+             cpu: 100m
+             memory: 128Mi
+         envFrom:
+         volumeMounts:
+           - name: fakeroot
+             mountPath: /fakeroot.pem
+             subPath: fakeroot.pem
+       restartPolicy: Never
+       volumes:
+         - name: fakeroot
+           configMap:
+             name: fakeroot
+   backoffLimit: 3
+   ttlSecondsAfterFinished: 86400
Upgrading release=jobs-demo, chart=../charts/jobs
Release "jobs-demo" has been upgraded. Happy Helming!
NAME: jobs-demo
LAST DEPLOYED: Tue Dec 22 11:27:28 2020
NAMESPACE: team-demo
STATUS: deployed
REVISION: 2
TEST SUITE: None
Listing releases matching ^jobs-demo$
jobs-demo	team-demo	2       	2020-12-22 11:27:28.365770446 +0000 UTC	deployed	jobs-0.1.0	0.1.0
UPDATED RELEASES:
NAME        CHART            VERSION
jobs-demo   ../charts/jobs     0.1.0
```

What do you mean by making "no sense to just run it on deployment"? It's obvious there are some use-cases for this. but if I understand correctly, this already covers "always deleting and starting a job fresh" or "when the spec changed".

Reply:
@Morriz Would you please take a look at this, then I can perform the requested changes and finish the PR.

Reply:
don't start work without requirements please...as always ;)

We want predictable outcomes for the following use cases:

- cronjob (easy, covered)
- always run upon deployment (can be realized by removing the job after it finished succesfuly)
- only run upon deployment when job changed (might need to realize it to "never remove", possibly with expiry of zero, but you have to assert that)

I don't think we need to write tests for these use cases as we would test for k8s doing its work

Reply:
So again: the `enabled` flag has nothing to do with this and should be removed

Reply:
> don't start work without requirements please...as always ;)

I haven't started anything, what do you mean exactly?

> always run upon deployment (can be realized by removing the job after it finished succesfuly)

Following your thoughts, that is what happens with the above command output. 

> only run upon deployment when job changed (might need to realize it to "never remove", possibly with expiry of zero, but you have to assert that)

> I don't think we need to write tests for these use cases as we would test for k8s doing its work

Again following your thoughts, you imply this case isn't covered right now, is this correct? Do you suggest to have at least a key where an user can specify what the deployment strategy will be? Something like:

```
deployment: always|onChange
```

The research I've done is that a job should only run once and not have any logic for re-running a job. It will be only possible by duplicating the old spec and rerunning it with a hacky kubectl bash command. I think it would be best to look into helm hooks.

> So again: the enabled flag has nothing to do with this and should be removed

You don't have to repeat yourself, but the fault is mine this time. The production ready version will be #258, unless it will be best to merge these branches. After this discussion I intend to close this pull request. I've already made the change there.

Reply:
I'll merge values-schema.yaml into this branch so we can agree on that after review.